src/general/IWorldEffect.cs

```diff
@@ -33,18 +34,17 @@ public void OnRegisterToWorld()
 
     public void OnTimePassed(double elapsed, double totalTimePassed)
     {
+        var glucose = SimulationParameters.Instance.GetCompound("glucose");
         foreach (var key in targetWorld.Map.Patches.Keys)
         {
             var patch = targetWorld.Map.Patches[key];
 
-            foreach (var compound in patch.Biome.Compounds.Keys)
+            // If there are microbes to be eating up the primordial soup, reduce the milk
+            if (patch.SpeciesInPatch.Count > 0)
             {
-                if (compound.InternalName == "glucose")
-                {
-                    var data = patch.Biome.Compounds[compound];
-
-                    data.Density *= Constants.GLUCOSE_REDUCTION_RATE;
-                }
+                var glucoseValue = patch.Biome.Compounds[glucose];
```

Comment:
I would use TryGetValue here just in case there might in the future be patches that don't contain the glucose compounds at all.